internal/admission/validator.go

```diff
@@ -81,7 +79,7 @@ func (validator KongHTTPValidator) ValidatePlugin(
 		config, err := kongstate.SecretToConfiguration(validator.Store,
 			k8sPlugin.ConfigFrom.SecretValue, k8sPlugin.Namespace)
 		if err != nil {
-			return false, fmt.Sprintf("could not load secret plugin configuration: %v", err), nil
+			return false, "could not load secret plugin configuration", err
```

Comment:
Consider creating `const`s for these error string literals, similar to lines 33-34